### DISCUSSION TITLE
Mpu6050 feature

### DISCUSSION THREAD

**Discussion by neergaby195** [MobiFlight/MobiFlight-Connector#1425] [Ideas]
Hello, is possible to add mpu6050 for axis because i have a modified saitek yoke and i use a mpu6050 for x axis and i discovered this project... and i like the ideea to implement.

**Comment by DocMoebiuz** [comment]
I am not sure I understand what you mean?

**Comment by tigert** [comment]
You could build this as USB HID device using maybe the RP2040 pico, or arduino pro micro. because a gyroscope / accelerometer unit will likely produce a lot of input data, and you likely wish to use it for something that needs to react fast.

USB joystick can be read as a usb joystick axis into MobiFlight if you want to do something with the orientation data of the sensor, but for things like head tracking or controller input, it might be best to route it directly via the USB HID layer for better performance in the sim.

That said, there is a plan to make it easier for users to modify the Mobiflight firmware to support custom hardware, so you can connect to MobiFlight to send and receive data from a flight simulator, while the rest of the firmware can be written to use whatever libraries and code is necessary to talk to the connected hardware. So it might be possible, but maybe a bit out of scope for the main project itself.

**Comment by tigert** [comment]
@neergaby195 actually, if you have a look at #1265 that would be another option in the future, but custom device support is for outputs only at this point, but it could be a thing to discuss whether it makes sense for inputs too, and then it would be possible to have the MPU work as a custom input device, but this is not yet beind developed, only at ideation stage.

**Comment by JaimeLeon2** [comment]
Food for thought, I just read the MPU6050 is at end of life and it is recommended it should  not be used in any new designs.

**Comment by tigert** [comment]
Yeah, and well, there are actually now ESP32 dev boards, cheap, that contain motion and orientation sensors. These would be a nice custom firmware project for some experimental or assistive controllers, since in MobiFlight you can program and try different interaction logics as well, in addition to just airplane logic. Like, for someone with disability in their mobility, turning their head and use that as an encoder like adjustment, for example.
